### PR TITLE
scannet dataset can pick the number of classes to use and referring e…

### PR DESCRIPTION
**Pull request by alexsax**
Issue:
- ScanNet dataset saved object classes in 3D using NYU40 classes.
- Referring expressions do not use these classes, instead using subsets of the full scannet classes. Te exact subset varies depending on scanrefer/referit3d

Fix:
This PR gets the classes for referring expressions by cross-referencing the target object id of the expression with the class names used by the dataset. 
Added ability to choose class naming + subsets in the dataset

Misc:
Some additional bugfixes for hydra configs in scannet_offline_eval to keep those up-to-date with InstanceMemory API

### CODE REVIEW

File: src/home_robot/home_robot/datasets/scannet/data/load_scannet_data.py [hunk 1]
```diff
@@ -180,6 +182,7 @@ def export(
                 instance_ids[verts] = object_id
                 if object_id not in object_id_to_label_id:
                     object_id_to_label_id[object_id] = label_ids[verts][0]
+            # print(object_id, object_id_to_label_id[object_id])
```
Comment (cpaxton) [inline src/home_robot/home_robot/datasets/scannet/data/load_scannet_data.py:185]: remove?

File: src/home_robot/home_robot/datasets/scannet/referit3d_data.py [hunk 2]
```diff
@@ -102,7 +105,7 @@ def load_referit3d_data(
         n_original = len(referit_data)
         referit_data = referit_data[referit_data["mentions_target_class"]]
         referit_data.reset_index(drop=True, inplace=True)
-        print(
+        logger.info(
```
Comment (cpaxton) [inline src/home_robot/home_robot/datasets/scannet/referit3d_data.py:108]: I like this, we should switch to using logger more

File: src/home_robot/home_robot/datasets/scannet/scannet_constants.py [hunk 3]
```diff
@@ -5602,7 +5603,7 @@
     "cd case",
     "closet rod",
     "coffee kettle",
-    "wardrobe ",
```
Comment (cpaxton) [inline src/home_robot/home_robot/datasets/scannet/scannet_constants.py:5605]: is this space important?

File: src/home_robot/home_robot/datasets/scannet/scannet_constants.py [hunk 4]
```diff
@@ -10411,6 +10412,564 @@
     1175,
 )
 
+
+# Equivalent to:
+# labels_pd = pd.read_csv('meta_data/scannetv2-labels.combined.tsv', sep='\t', header=0)
+# labels_pd.loc[labels_pd.raw_category == 'stick', ['category']] = 'object'
+# labels_pd.loc[labels_pd.category == 'wardrobe ', ['category']] = 'wardrobe'
+# CLASS_ID_TO_NAME = dict(zip(labels_pd['id'], labels_pd['category']))
+CLASS_ID_TO_NAME = {
```
Comment (cpaxton) [inline src/home_robot/home_robot/datasets/scannet/scannet_constants.py:10421]: Should we point to a specific scannet version to make sure this stays in sync? Not that i expect it to change or anything

**Review by cpaxton — APPROVED** [review]
lgtm

File: src/home_robot/home_robot/datasets/scannet/scannet_constants.py [hunk 5]
```diff
@@ -1016,7 +1016,8 @@
     "cd case",
     "closet rod",
     "coffee kettle",
-    "wardrobe ",
+    # "wardrobe ",
```
[the remark this answers is not in the document]
Comment (alexsax) [reply src/home_robot/home_robot/datasets/scannet/scannet_constants.py:1019]: Somehow the original scannet category has this trailing whitespace. It screws up some code that uses string-based matching :/

File: src/home_robot/home_robot/datasets/scannet/scannet_constants.py (hunk 4 above)
Comment (alexsax) [reply src/home_robot/home_robot/datasets/scannet/scannet_constants.py:10421]: Thing that changes seems to be the number of classes, and whether those classes come from the scannet annotators or else MP40 or NYU40 classes. That mapping is in this [csv](https://github.com/daveredrum/ScanRefer/blob/9d7483053e8d29acfd4db4eb1bc28f1564f5dddb/data/scannet/meta_data/scannetv2-labels.combined.tsv), and we might want to restrict our attention to some subset of the full scannet categories (that's what this constants file is supposed to do). 

Eventually we probably want to pin the evaluation config for the dataset (e.g. frame skip, image size, etc) but I think we can do that in configs

**Pull request merged**